### PR TITLE
Fix: Add extract_references to TypedTableBlock for media reference tracking

### PR DESCRIPTION
**Pull request by amxnn05**
## Issue [#13581](https://github.com/wagtail/wagtail/issues/13581)
Media items (Images and Documents) embedded in RichTextBlock cells within a TypedTableBlock were not being indexed in Wagtail's reference tracking system. This prevented editors from seeing usage information and could lead to data loss during cleanup operations.
## Root Cause
The `BaseTypedTableBlock` class was missing the `extract_references` method that the reference indexing system (`ReferenceIndex._extract_references_from_object`) relies on.
## Changes
- Added `extract_references` method to `BaseTypedTableBlock` class
- Method iterates through table cells and delegates to child blocks
- Properly formats paths with row/column information
- Follows the same pattern as `StructBlock` and `ListBlock`
## Testing
- Added `test_extract_references` - verifies PageChooserBlock references are extracted
- Added `test_extract_references_empty_table` - ensures empty tables handled gracefully

### CODE REVIEW

**Comment by p-r-a-v-i-n** [comment]
If took help from AI please mention in the PR as per contributing guidelines.

**Comment by amxnn05** [comment]
ok,  i have updated that.

**Comment by gasman** [comment]
> I reverted changes and only one test case was failing

@p-r-a-v-i-n Assuming it's `test_extract_references_empty_table` that passes in both cases - I think that's fine. When `extract_references` was a no-op, it would have trivially returned no results for an empty table - confirming that it still behaves that way when there _is_ meaningful logic in `extract_references` that could trip up on that case is valuable information.

**Comment by p-r-a-v-i-n** [comment]
Hi @amxnn05 , 
Thank you so much for taking note of my feedback I really appreciate that. Though I'm not the right person to review this PR.
I just wanted to bring your attention to the test case if those are working as you were expecting or not.

@gasman thanks that you looked through it. I may be completely wrong here but I feel if  that test was fine in both cases then it won't make sense adding it. There has to be failing test case

File: wagtail/contrib/typed_table_block/tests.py [hunk 1]
```diff
@@ -372,6 +372,114 @@ def test_get_searchable_content_includes_caption_headings_and_cells(self):
...
+        block = TypedTableBlock(
+            [
+                ("rich_text", blocks.RichTextBlock()),
+                ("page", blocks.PageChooserBlock()),
+            ]
+        )
+
+        # Create test data with embedded image in rich text and page reference
+        # Rich text with embedded image (id=123)
+        rich_text_html = '<p>Some text <embed embedtype="image" id="123" format="left" alt="Test image" /> more text</p>'
```
Comment (gasman) [inline wagtail/contrib/typed_table_block/tests.py:401]: It's a little odd to be testing this with a made-up image ID, rather than one that we've created - but we do in fact want 'orphaned' image references to be treated the same as live ones, and this detail isn't important for the current test (we're not really interested in how the references arise within the cell blocks, only that they get propagated by the TypedTableBlock). So, I'll accept this...

**Comment by gasman** [comment]
Merged in 3ae5870c81b332b6807d30d722bd0b8cf214dd26

**Comment by gasman** [comment]
@p-r-a-v-i-n If it was the _only_ test being added, then yes, it would be insufficient - but alongside the test for "normal working" it's a useful test coverage for an edge case that wasn't being considered before. It's the equivalent of someone building electric bikes and having a test to confirm that it doesn't set on fire - sure, it didn't set on fire before you added the electrics, but that was never even considered a possibility before that point. It's still a worthwhile test :-)

**Comment by p-r-a-v-i-n** [comment]
@gasman I think it's debatable. Electric bikes, bridges, cars , houses don't have that advantage what building software has.  We can produce the failure before it produced at the user end electric bikes can't afford this .

If we added this test that this has to be failed somewhere ? or what are the scenarios where it will failed or currently we don't any scenario? If yes then where is the test for it ?

**Comment by gasman** [comment]
@p-r-a-v-i-n The purpose of unit tests is to define what we consider to be correct behaviour of the application, so that if we make a change that breaks it in future, it will be flagged up immediately without relying on us having to catch it in manual testing. In this specific case, the old code worked accidentally: given an empty table, it returned an empty list of references, but only because it returned an empty list of references in _all_ cases.

For any bugfix or new feature, we would expect the author to provide at least one test that failed against the old code (because otherwise, that would mean that we hadn't identified a single thing that the new code does better). There's no rule to say that _all_ tests have to be previous failures.

In this case, I'm going to guess that @amxnn05 initially left out the lines

        if not value:
            return

while developing their fix, but then found when testing in the browser that this crashed on an empty table. They rightly decided that we shouldn't rely on manual testing to catch this bug, and added a test to cover it.

**Comment by p-r-a-v-i-n** [comment]
@gasman 
Its seems like you did not understand my point.  My point is keep the second test. but was there any scenario where it might fail ? if yes, then  shouldn't we add test to cover it?

> There's no rule to say that all tests have to be previous failures.

I think that's a common practice. Irrelevant test would be garbage.

> while developing their fix, but then found when testing in the browser that this crashed on an empty table. They rightly decided that we shouldn't rely on manual testing to catch this bug, and added a test to cover it.

I didn't get what you mean here. 

> then found when testing in the browser that this crashed on an empty table.

So is it crashing now ? Bcs test don't take any part in stopping that crash.

> we shouldn't rely on manual testing to catch this bug, and added a test to cover it.

what does it covering ? if removed , what will fail ?

**Comment by laymonage** [comment]
@p-r-a-v-i-n 

> I didn't get what you mean here.

See it from the other way around: the test is added to ensure the **new** code can handle that empty case. Specifically this part in the implementation:

```py
        if not value:
            return
```

Without that branch, the new code would've raised an error and the new test would've failed.

The new test wouldn't fail with the old code because the feature was not implemented at all.

If you remove the test with the new code, nothing would fail, but we would have no test that ensures `extract_references` can handle the case when the table is empty. If in the future that `if` block were removed for any reason, we wouldn't have a failing test case to notify us. That's the purpose of the test.
